src/LitJson/JsonMapper.cs

```diff
@@ -782,7 +782,7 @@ private static void RegisterBaseImporters ()
             if (obj is IDictionary) {
                 writer.WriteObjectStart ();
                 foreach (DictionaryEntry entry in (IDictionary) obj) {
-                    writer.WritePropertyName ((string) entry.Key);
+                    writer.WritePropertyName (entry.Key.ToString());
```

Comment:
To not get any culture issues when serializing yes, we should use invariant culture, something like 
```csharp
            if (obj is IDictionary dictionary) {
                writer.WriteObjectStart ();
                foreach (DictionaryEntry entry in dictionary) {
                    var propertyName = entry.Key is string key
                        ? key
                        : string.Format(CultureInfo.InvariantCulture, "{0}", entry.Key);

                    writer.WritePropertyName (propertyName);
                    WriteValue (entry.Value, writer, writer_is_private,
                                depth + 1);
                }
                writer.WriteObjectEnd ();

                return;
            }
```

Reply:
For perf it should me `Convert.ToString(entry.Key, CultureInfo.InvariantCulture);` and not `string format`.

Reply:
I have made the changes you requested